conformance/core/empty_document.ion

```diff
@@ -0,0 +1,55 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+// Simple test-cases for bootstrapping the conformance DSL.
+
+(document (denotes))
+(document (produces))
+
+(document null.string (denotes))
```

Comment:
I found that accepting `null.string` was natural, since the implementation needs a way to signal "no text" so this was a perfect sentinel value.  It's also just another exception case to document, and we have no established way to test that behavior anyway.